diff --git a/dist/index.cjs b/dist/index.cjs
index 44601f5a58a8d28f1f387a24419b0867fbbeb8cd..af4a3b5d084dbe3354a92e5a1a9faf653ee2e091 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -195,7 +195,9 @@ function generateLqipClassName(resource) {
   if (generatedClassNames.has(resource)) {
     return generatedClassNames.get(resource);
   } else {
-    const className = `ri-dyn-${b64.encode(generatedClassNames.size)}`;
+    const hash = node_crypto.createHash('md5').update(resource).digest();
+    const num = hash.readUInt32LE(0);
+    const className = `ri-dyn-${b64.encode(num)}`;
     generatedClassNames.set(resource, className);
     return className;
   }
diff --git a/dist/index.mjs b/dist/index.mjs
index ddeb682a542a8fa49be7b1902c8fb110caeffa5d..28363f3b9caa798754b94336e006fa0164e4355a 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -193,7 +193,9 @@ function generateLqipClassName(resource) {
   if (generatedClassNames.has(resource)) {
     return generatedClassNames.get(resource);
   } else {
-    const className = `ri-dyn-${b64.encode(generatedClassNames.size)}`;
+    const hash = createHash('md5').update(resource).digest();
+    const num = hash.readUInt32LE(0);
+    const className = `ri-dyn-${b64.encode(num)}`;
     generatedClassNames.set(resource, className);
     return className;
   }
